### ISSUE TITLE
Initially failed to run

### ISSUE THREAD

**Issue by dustingraham** [hawkeye64/electron-quasar-file-explorer#1]
Using fedora 29. Downloaded, ran npm install, tried npm run dev, crashed with the following.

``` ERROR  Failed to compile with 4 errors                                                                                                                                                                                                                                    9:19:21 AM

Module build failed (from ./node_modules/vue-loader/lib/index.js):
  Error: [vue-loader] vue-template-compiler must be installed as a peer dependency, or a compatible compiler implementation must be passed via options.
  
  - index.js:21 loadTemplateCompiler
    [electron-quasar-file-explorer-master]/[vue-loader]/lib/index.js:21:11
  
  - index.js:65 Object.module.exports
    [electron-quasar-file-explorer-master]/[vue-loader]/lib/index.js:65:35
  
```

Ran `npm i vue-loader` still didin't work.

Tried `npm i vueify` still didn't work.

Tried `npm i vue-template-compiler`

And, viola, it's working. Clicking around is responsive. I see thumbnails. Cool. Thank you for the tutorial!

**Comment by hawkeye64** [comment]
`npm install --dev vue-template-compiler` should fix your issue.
I'll be taking a look at this on the weekend.

**Comment by dustingraham** [comment]
Cool, as I mentioned, I got it working. I just wanted to provide this feedback because as a new user to quasar, I expected the demo to work without manually adding a package. Feel free to close if you don't want to make any changes.

**Comment by hawkeye64** [comment]
I pulled a new clone and built without issue. I believe the problem here was that you likely used `npm` to build and we have definitive proof that `npm` has issues with local packages and dependencies. It is recommended to install `yarn`, remove your `node_modules` folder, run `yarn`, then run `yarn dev`.

Would you be so helpful and try out these steps? Thanks.

**Comment by dustingraham** [comment]
Confirm.

```
git clone https://github.com/hawkeye64/electron-quasar-file-explorer.git
cd electron-quasar-file-explorer
yarn
yarn dev
```

Works.

```
git clone https://github.com/hawkeye64/electron-quasar-file-explorer.git
cd electron-quasar-file-explorer
npm install
npm run dev
```
Fails with the vue-template-compiler error.

As mentioned, running `npm install vue-template-compiler` allows it to work.

```diff
$ git diff
diff --git a/package.json b/package.json
index 235dbff..36af0d1 100644
--- a/package.json
+++ b/package.json
@@ -23,7 +23,8 @@
     "express": "^4.16.4",
     "mime-types": "^2.1.21",
     "moment": "^2.22.2",
-    "pretty-bytes": "^5.1.0"
+    "pretty-bytes": "^5.1.0",
+    "vue-template-compiler": "^2.6.10"
   },
   "devDependencies": {
     "babel-eslint": "^8.2.1",
```

Any drawback to adding that so that it works with NPM? Personally I prefer npm over yarn.

I've already learned a lot from what you provided, so thanks. I don't need this to work for myself. Just thought it'd be nice if any future users didn't stumble over this.

Thanks.

**Comment by hawkeye64** [comment]
I have run intoi so many issues with NPM. I have learned to use NPM for globals (because I use NVM) and Yarn for locals. Have seen a LOT of issues with NPM on Windows as well.

**Comment by hawkeye64** [comment]
So, did this:
```
  "peerDependencies": {
    "vue-template-compiler": "^2.6.10"
  }
```
and got this:
```
Module build failed (from ./node_modules/vue-loader/lib/index.js):
  Error: [vue-loader] vue-template-compiler must be installed as a peer dependency, or a compatible compiler implementation must be passed via options.
  
  - index.js:21 loadTemplateCompiler
    [electron-quasar-file-explorer]/[vue-loader]/lib/index.js:21:11
  
  - index.js:65 Object.module.exports
    [electron-quasar-file-explorer]/[vue-loader]/lib/index.js:65:35
```
Even tried putting it in devDependencies and same issue. This is with yarn.

with npm:

```
ERROR in ./src/pages/Error404.vue
Module build failed (from ./node_modules/vue-loader/lib/index.js):
  Error: [vue-loader] vue-template-compiler must be installed as a peer dependency, or a compatible compiler implementation must be passed via options.
```

Open for suggestions?

**Comment by hawkeye64** [comment]
btw, witthout adding it `npm run dev` works for me. As does `yarn dev`.

**Comment by dustingraham** [comment]
In that case, you might have vue-template-compiler installed in your global npm repo.

I use node a lot. Generally linux, but sometimes on windows. I don't have many issues. I'll say that I have _seen_ issues due to packages installed globally versus locally.

Anyways, I'll close this. All good.

**Comment by hawkeye64** [comment]
It's not global, Quasar has it in the chain, but NPM is having a hard time seeing it.

**Comment by dustingraham** [comment]
I'm on linux, using npm version 6.9.0. A fresh clone of the repo fails to work for me on 3 different workstations. Adding vue-template-compiler to dependencies or devDependencies allows it to work for me.

I don't know why yours is working without it. You could try blowing away your node modules to see if you can reproduce the failure. You could also try on a different machine with a fresh checkout and running npm install followed by npm run dev.

**Comment by dustingraham** [comment]
(regarding your comment, "btw, witthout adding it `npm run dev` works for me. As does `yarn dev`." it wasn't clear whether you deleted your node_modules prior to each test. It could be that running yarn pulls in the peer dependency automatically, then since it's already there, your npm run dev test is not a true test of how an npm user might approach this example since the library is already downloaded to node modules.)

**Comment by hawkeye64** [comment]
I'm on vacation in Wales for the next two weeks. Can you put a PR together and I'll accept it. Thanks. On Apr. 30, 2019 7:05 p.m., Dustin Graham <notifications@github.com> wrote:Closed #1.
